pkg/test/environment.go

```diff
@@ -529,31 +529,65 @@ func (e *Environment) startRealtimeServer() {
 	go srv.Start(e.ctx)
 }
 
+type PrometheusJob struct {
+	JobName    string
+	ScrapePort int
+}
+
 type prometheusTemplateOptions struct {
 	ListenPort    int
 	RTMetricsPort int
 	OpniAgentPort int
+	// these fill in a text/template defined as {{.range Jobs}} /* */ {{end}}
+	Jobs []PrometheusJob
+}
+
+type additionalPrometheusConfig struct {
+	configContents string
+	jobs           []PrometheusJob
+}
+
+func NewAdditionalPrometheusConfig(configPath string, jobs []PrometheusJob) *additionalPrometheusConfig {
+	return &additionalPrometheusConfig{
+		configContents: string(TestData(configPath)),
+		jobs:           jobs,
+	}
 }
 
-func (e *Environment) StartPrometheus(opniAgentPort int) int {
+// `prometheus/config.yaml` is the default monitoring config.
+// `slo/prometheus/config.yaml` is the default SLO config.
+func (e *Environment) StartPrometheus(opniAgentPort int, config *additionalPrometheusConfig) int {
```

Comment:
this should use the options pattern such that this would be called like 
```
env.StartPrometheus(port)
// or
env.StartPrometheus(port, WithConfig(...))
// or
env.StartPrometheus(port, WithConfig(...), WithJobs(...))
// ...etc
```